synopsys-task/src/main.ts

```diff
@@ -72,19 +85,72 @@ export async function run() {
   console.log("Synopsys Task workflow execution completed");
 }
 
-export function logBridgeExitCodes(message: string): string {
-  const exitCode = message.trim().slice(-1);
+export function getExitMessage(message: string, exitCode: string): string {
   return constants.EXIT_CODE_MAP.has(exitCode)
-    ? "Exit Code: " + exitCode + " " + constants.EXIT_CODE_MAP.get(exitCode)
-    : message;
+    ? "Exit Code: " + exitCode + " - " + constants.EXIT_CODE_MAP.get(exitCode)
+    : "Exit Code: " +
+        ErrorCode.UNDEFINED_ERROR_FROM_EXTENSION.toString() +
+        " - " +
+        "Undefined error from extension: " +
+        message;
 }
 
-run().catch((error) => {
-  if (error.message != undefined) {
-    taskLib.error(error.message);
+export function getStatusFromError(errorObject: Error): string {
+  return errorObject.message.trim().split(" ").pop() || "";
+}
+
+function markBuildStatusIfIssuesArePresent(
+  status: string,
+  taskResult: TaskResult,
+  errorMessage: string
+) {
+  const exitMessage = getExitMessage(errorMessage, status);
+
+  if (status == ErrorCode.BRIDGE_BREAK_ENABLED.toString()) {
+    console.log(errorMessage);
+    if (taskResult === TaskResult.Succeeded) {
+      console.log(exitMessage);
+    }
+    console.log(
+      `Marking the build ${TaskResult[taskResult]} as configured in the task`
+    );
+    taskLib.setResult(taskResult, exitMessage);
+  } else {
+    taskLib.error(errorMessage);
+    console.log(
+      `Marking build status ${TaskResult[taskResult]} is ignored since exit code is: ${status}`
+    );
     taskLib.setResult(
       taskLib.TaskResult.Failed,
-      "Workflow failed! ".concat(logBridgeExitCodes(error.message))
+      "Workflow failed! ".concat(exitMessage)
     );
   }
+}
+
+run().catch((error) => {
+  if (error.message != undefined) {
+    const isReturnStatusEnabled = parseToBoolean(inputs.RETURN_STATUS);
+    const status = getStatusFromError(error);
+
+    // The statement set the exit code in the 'status' variable which can be used in the YAML file
+    if (isReturnStatusEnabled) {
+      console.log(
+        `##vso[task.setvariable variable=status;isoutput=true]${status}`
+      );
+    }
+
+    const taskResult: TaskResult | undefined = getMappedTaskResult(
+      inputs.MARK_BUILD_STATUS
+    );
+
+    if (taskResult !== undefined && taskResult !== TaskResult.Failed) {
```

Comment:
There is a reason I am explicitly checking undefined, because if you look into ADO `TaskResult` enum returns integer values like - 0,1,2,.. 

This is the structure for TaskResult enum - 
```
export declare enum TaskResult {
    Succeeded = 0,
    SucceededWithIssues = 1,
    Failed = 2,
    Cancelled = 3,
    Skipped = 4
}
```
 I need to the use "0" value too to mark the build Succeeded. But if I check like this - 
`if (taskResult)` then `Succeeded=0` is getting ignored since 0 is considered as `false`. Therefore in `getMappedTaskResult()` method, I am returning `undefined` explicitly and here I am checking for `undefined` value only.